src/imageops/filter_1d.rs

```diff
@@ -0,0 +1,879 @@
+        new_kernel.remove(new_kernel.len() - 1);
+    }
+
+    new_kernel
+}
+
+const RING_QUEUE_CIRCULAR_CUTOFF: usize = 55;
+
+fn filter_1d_ring_queue<T, F, I, const N: usize>(
+    image: &[T],
+    destination: &mut [T],
+    image_size: FilterImageSize,
+    row_kernel: &[F],
+    column_kernel: &[F],
+) -> Result<(), ImageError>
+where
+    T: Copy + AsPrimitive<F> + Default + Send + Sync + KernelTransformer<F, I> + AsPrimitive<I>,
+    F: Default + 'static + Copy,
+    I: ToStorage<T>
+        + Mul<I, Output = I>
+        + Add<I, Output = I>
+        + MulAdd<I, Output = I>
+        + Send
+        + Sync
+        + PartialEq
+        + Default
+        + 'static
+        + Copy,
+    i32: AsPrimitive<F> + AsPrimitive<I>,
+    f64: AsPrimitive<T>,
+{
+    if image.len() != image_size.width * image_size.height * N {
+        return Err(ImageError::Limits(LimitError::from_kind(
+            LimitErrorKind::DimensionError,
+        )));
+    }
+    if destination.len() != image_size.width * image_size.height * N {
+        return Err(ImageError::Limits(LimitError::from_kind(
+            LimitErrorKind::DimensionError,
+        )));
+    }
+
+    let mut scanned_row_kernel = row_kernel
+        .iter()
+        .map(|&x| T::transform(x))
+        .collect::<Vec<I>>();
+    let mut scanned_column_kernel = column_kernel
+        .iter()
+        .map(|&x| T::transform(x))
+        .collect::<Vec<I>>();
+
+    scanned_row_kernel = prepare_symmetric_kernel(&scanned_row_kernel);
+    scanned_column_kernel = prepare_symmetric_kernel(&scanned_column_kernel);
+
+    if scanned_row_kernel.is_empty() || scanned_column_kernel.is_empty() {
+        for (dst, src) in destination.iter_mut().zip(image.iter()) {
+            *dst = *src;
+        }
+        return Ok(());
+    }
+
+    let pad_w = (scanned_row_kernel.len() / 2).max(1);
+
+    let arena_width = image_size.width * N + pad_w * 2 * N;
+    let mut row_buffer = vec![T::default(); arena_width];
+
+    // If column kernel is small better to use ring queue circular convolution
+
+    let full_width = image_size.width * N;
+
+    // Flat ring queue
+    let mut buffer = vec![T::default(); image_size.width * N * scanned_column_kernel.len()];
+
+    let column_kernel_len = scanned_column_kernel.len();
+
+    let half_kernel = column_kernel_len / 2;
+
+    make_arena_row::<T, N>(
+        image,
+        &mut row_buffer,
+        0,
+        image_size,
+        KernelShape {
+            width: scanned_row_kernel.len(),
+            height: 0,
+        },
+    );
+    filter_symmetric_row::<T, I, N>(&row_buffer, &mut buffer[..full_width], &scanned_row_kernel);
+
+    let (src_row, rest) = buffer.split_at_mut(full_width);
+    for dst in rest.chunks_exact_mut(full_width).take(half_kernel) {
+        for (dst, src) in dst.iter_mut().zip(src_row.iter()) {
+            *dst = *src;
+        }
+    }
+
+    let mut start_ky = column_kernel_len / 2 + 1;
+
+    start_ky %= column_kernel_len;
```

Comment:
This one is odd. It just catches the case of `column_kernel_len == 1` but that rather seems like a very special case on its own which doesn't need column buffers at all to be honest.

Reply:
I was just confused by the way those lines are written. It suggests that `column_kernel_len == 1` is an even more special case than anisotropy itself since no other case required the modulos operation and the length of 1 does not really require any intermediate buffers.